### ISSUE TITLE
Getting Automatic instead of Hex code for Chart Fill / Line color

### ISSUE THREAD

**Issue by ExcelOffTheGrid** [OfficeDev/office-js#2497]
On an Excel chart I'm trying to match the data label to the fill (column chart) / line color (line chart) from button click in JS Add-in panel.

The solidColor property is not currently available in the API. Jakob Nielsen pointed me to the Preview Libraries:
https://appsforoffice.microsoft.com/lib/beta/hosted/office.js
https://appsforoffice.microsoft.com/lib/beta/hosted/office.d.ts

However, when using the new solidColor property the value returned is a Hex code, unless the color is the default color. In which case the value returned is "Automatic". Which isn't helpful, as I can't change the text color of the data label to "Automatic". Even if we could Automatic for a Chart is not the same Automatic color for text.

## Expected Behavior
I would expect the Hex color code returned in all instances. Or there should be an easy way to get the Hex code from "Automatic"

## Current Behavior
The color code returned shows as "Automatic"

## Steps to Reproduce, or Live Example
Code snippet from ScriptLab:

`let workbook = context.workbook;`
`let chart = workbook.getActiveChart();`
`let newSeries = chart.series;`
`let series0 = newSeries.getItemAt(0);`
`series0.load("format/fill");`
`await context.sync();`
`let color = series0.format.fill;`
`console.log(color["solidColor"]);`

**Comment by davidchesnut** [comment]
Hi @ExcelOffTheGrid,
Thanks for letting us know about this issue! @grangeryy, is this something you can help with?

Thanks!
David

**Comment by grangeryy** [comment]
@MiaofeiWang could you help to take a look at this? THX!

**Comment by grangeryy** [comment]
Hey @ExcelOffTheGrid, we did initial investigation on this. The behavior to return "Automatic" was a by design in the past. We need to change the design/add new API to return Hex color code. Backlog item created to track this ask:5933668

**Comment by ExcelOffTheGrid** [comment]
Thanks @grangeryy 

I'm surprised that "Automatic" is by design. 

The VBA equivalent of this returns the long color code while the Office JS shows Automatic.  Look at the screen shot below, while it is not Hex, it still provides an exact color code that we can use.

![ColorCodeComparison](https://user-images.githubusercontent.com/83125193/160918681-3e3bcbad-04e0-48b9-9eff-9526538c85c6.png)

VBA Code:
```
Dim cht As Chart
Set cht = ActiveChart
MsgBox cht.SeriesCollection(1).Format.Fill.ForeColor
```

Out of interest, what does "Automatic" mean? Look at the screenshot below, after selecting a chart fill, if I select the fill palette, it shows the "Automatic" color as being White (#FFFFFF), but the color in the chart which Office JS has determined as "Automatic" is Blue.

![Automatic Fill Color](https://user-images.githubusercontent.com/83125193/160917661-4a2d7b32-dd0d-427e-b704-942d9ca37716.png)

So, even within the same instance of Excel "Automatic" != "Automatic"

**Comment by MiaofeiWang** [comment]
Hi @ExcelOffTheGrid , "Automatic" is the option shown when you open the formating pane.
![image](https://user-images.githubusercontent.com/28795134/160962586-99817125-dad7-4907-87d6-d7429366ad0f.png)
The actual color for "Automatic" changes if the theme or theme color changes. But somehow you are right that seems the concept is confusing with the "Automatic" in color pane. 

<img width="684" alt="image" src="https://user-images.githubusercontent.com/28795134/160962776-97f49712-f3b1-4403-b54f-4433883b920c.png">

**Comment by ExcelOffTheGrid** [comment]
OK, I see where that comes from.

In which case then, it gets even more confusing because "Automatic" could be any color, and even varies by chart series.

This illustrates even more why this property should return a Hex code. rather than "Automatic".

**Comment by MiaofeiWang** [comment]
@ExcelOffTheGrid We totally got your point. 
As this API is still in preview stage, we will discuss whether changing the design to always returning a hex code. If so, I am afraid this API won't be able to go to public in next 2 or 3 months. 
Thanks for your suggestion!

**Comment by MiaofeiWang** [comment]
Hi @ExcelOffTheGrid , great thanks for the suggestion.
After our internal discussion, the new API `ChartFill.getSolidColor()` will always return the hex code of color. It was released in Excel api set 1.16 recently and here is the document: https://learn.microsoft.com/en-us/javascript/api/excel/excel.chartfill?view=excel-js-preview#excel-excel-chartfill-getsolidcolor-member(1).
